ledger-core/virtual/lmn/register.go

```diff
@@ -106,178 +45,54 @@ type SubSMRegister struct {
 	IncomingResult    *execution.Update
 	Interference      isolation.InterferenceFlag
 	ObjectSharedState object.SharedStateAccessor
-	DryRun            bool
 
-	Object          reference.Global
-	LastFilamentRef reference.Global
-	LastLifelineRef reference.Global
-
-	SafeResponseCounter smachine.SharedDataLink
-	PulseNumber         pulse.Number
+	Object      reference.Global
+	PulseNumber pulse.Number
 
 	// internal data
-	messages     MessagesHolder
-	requiredSafe int
-	sendError    error
 
 	// output arguments
-	NewObjectRef       reference.Global
-	NewLastFilamentRef reference.Global
-	NewLastLifelineRef reference.Global
+	Context       *RegistrationCtx
+	lastBranchRef reference.Global
+	lastTrunkRef  reference.Global
+	Messages      []SerializableBasicMessage
 
 	// DI
-	messageSender messageSenderAdapter.MessageSender
-	pulseSlot     *conveyor.PulseSlot
-	refBuilder    RecordReferenceBuilderService
-}
-
-var dSubSMRegisterInstance smachine.StateMachineDeclaration = &dSubSMRegister{}
-
-type dSubSMRegister struct {
-	smachine.StateMachineDeclTemplate
-}
-
-func (*dSubSMRegister) InjectDependencies(sm smachine.StateMachine, _ smachine.SlotLink, injector injector.DependencyInjector) {
-	s := sm.(*SubSMRegister)
-
-	injector.MustInject(&s.messageSender)
-	injector.MustInject(&s.pulseSlot)
-	injector.MustInject(&s.refBuilder)
-}
-
-func (*dSubSMRegister) GetInitStateFor(_ smachine.StateMachine) smachine.InitFunc {
-	return nil
-}
-
-func (s *SubSMRegister) GetSubroutineInitState(ctx smachine.SubroutineStartContext) smachine.InitFunc {
-	ctx.SetSubroutineCleanupMode(smachine.SubroutineCleanupAliasesAndShares)
-
-	return s.Init
-}
-
-func (s *SubSMRegister) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
-	return dSubSMRegisterInstance
+	PulseSlot        *conveyor.PulseSlot
```

Comment:
I'd suggest to isolate this builder from PulseSlot due to unnecessary specifics.

Reply:
fixed